compiler/rustc_hir_analysis/src/check/intrinsic.rs

```diff
@@ -106,14 +111,25 @@ pub fn intrinsic_operation_unsafety(intrinsic: Symbol) -> hir::Unsafety {
         | sym::variant_count
         | sym::ptr_mask => hir::Unsafety::Normal,
         _ => hir::Unsafety::Unsafe,
+    };
+
+    if has_safe_attr != is_in_list {
+        tcx.sess.err(
+            DiagnosticMessage::Str(format!(
+                    "intrinsic safety mismatch between list of intrinsics within the compiler and core library intrinsics for intrinsic `{}`",
+                    tcx.item_name(intrinsic_id)
+        )));
```

Comment:
Ah, that's my bad. I should have pointed you toward `tcx.sess.span_err`. Sorry about that! 

I think this will generate a reasonable span for the error:

```suggestion
        tcx.sess.span_err(
            tcx.def_span(intrinsic_id),
            DiagnosticMessage::Str(format!(
                    "intrinsic safety mismatch between list of intrinsics within the compiler and core library intrinsics for intrinsic `{}`",
                    tcx.item_name(intrinsic_id)
        )));
```

Reply:
You're welcome! 😄

Reply:
Done in [37bf8f8](https://github.com/rust-lang/rust/pull/100719/commits/37bf8f888cb7b8a7cb461f0d41588e500b0bb9d1). Thanks again :)